wasm/libs/default_executor/src/multisig/account.rs

```diff
@@ -0,0 +1,96 @@
+//! Validation and execution logic of instructions for multisig accounts
+
+use super::*;
+
+impl VisitExecute for MultisigRegister {
+    fn visit(&self, executor: &mut Executor) {
+        let host = executor.host();
+        let target_domain = self.account.domain();
+
+        // Any account in a domain can register any multisig account in the domain
+        // TODO Restrict access to the multisig signatories?
+        // TODO Impose proposal and approval process?
+        if target_domain != executor.context().authority.domain() {
+            deny!(
+                executor,
+                "multisig account and its registrant must be in the same domain"
+            )
+        }
+
+        let Ok(_domain_found) = host
+            .query(FindDomains)
+            .filter_with(|domain| domain.id.eq(target_domain.clone()))
+            .execute_single()
+        else {
```

Comment:
```suggestion
        if !host
            .query(FindDomains)
            .filter_with(|domain| domain.id.eq(target_domain.clone()))
            .execute_single().is_ok() {
```

usually clippy tells you this

Reply:
also, this check is unnecessary. How about you add `debug_assertions`?

Reply:
you shouldn't `deny` in this case. Proper thing to do here is `panic`